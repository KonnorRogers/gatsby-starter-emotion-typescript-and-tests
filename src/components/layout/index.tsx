/**
 * Layout component that queries for data
 * with Gatsby's useStaticQuery component
 *
 * See: https://www.gatsbyjs.org/docs/use-static-query/
 */

import React from "react";
import PropTypes from "prop-types";
import { useStaticQuery, graphql } from "gatsby";
import { Global } from "@emotion/core";

import Header from "../header";
import * as styles from "./styles";

export interface Props extends React.HTMLAttributes<HTMLElement> {
  children?: React.ReactNode;
  data: {
    site: {
      siteMetadata: {
        title: string;
      };
    };
  };
}

export const PureLayout: React.FC<Props> = ({ children, data, ...rest }) => {
  return (
    <div {...rest}>
      <Global styles={styles.layout} />
      <Header siteTitle={data.site.siteMetadata.title} />
      <div
        style={{
          margin: `0 auto`,
          maxWidth: 960,
          padding: `0 1.0875rem 1.45rem`,
        }}
      >
        <main>{children}</main>
        <footer>
          Â© {new Date().getFullYear()}, Built with
          {` `}
          <a href="https://www.gatsbyjs.org">Gatsby</a>
        </footer>
      </div>
    </div>
  );
};

const Layout: React.FC<Props> = ({ children, ...rest }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return (
    <PureLayout data={data} {...rest}>
      {children}
    </PureLayout>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
