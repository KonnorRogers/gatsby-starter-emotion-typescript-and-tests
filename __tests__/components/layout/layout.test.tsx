import React from "react";
import { render } from "@testing-library/react";

import data from "Fixtures/layout-fixture";
import { PureLayout as Layout } from "~components/layout";

describe("Layout component", () => {
  test("Should render without error", () => {
    const { asFragment, getByText, getByTestId } = render(
      <Layout data={data} data-testid="layout" />
    );

    expect(getByTestId(/layout/)).toBeTruthy();
    expect(getByText(/Test Title/)).toBeTruthy();
    expect(asFragment).toMatchSnapshot();
  });
});
