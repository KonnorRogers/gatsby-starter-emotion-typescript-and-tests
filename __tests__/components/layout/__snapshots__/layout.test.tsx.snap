// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component Should render without error 1`] = `[Function]`;
